### ISSUE TITLE
Rename 'controller' to 'strategy'

### ISSUE THREAD

**Issue by jpkrohling** [jaegertracing/jaeger-operator#123]
The new Operator SDK has a notion of Controllers, which conflicts with what we had before. A new name for our controllers could be `Strategy`, so, this issue is about moving our old controllers to this new package.

**Comment by jpkrohling** [comment]
This is currently blocked by #122 but the branch is ready: https://github.com/jpkrohling/jaeger-operator/tree/123-Rename-controllers-to-strategy

**Comment by jpkrohling** [comment]
I believe this was already done.
